Apps/Playground/iOS/ViewController.swift

```diff
@@ -29,26 +27,24 @@ class UIBabylonGestureRecognizer: UIGestureRecognizer {
         super.touchesBegan(touches, with: event)
         
         for touch in touches {
-            let deviceSlot = _activeTouchIds.firstIndex(of: -1)
-            
-            if (deviceSlot != nil) {
-                _activeTouchIds[deviceSlot!] = touch.hash
-                let loc = touch.location(in: view)
-                
-                _onTouchDown(Int32(deviceSlot!), Int32(loc.x), Int32(loc.y))
-            }
+          if let deviceSlot = _activeTouchIds.firstIndex(of: -1) {
```

Comment:
nit: generally considered better practice to use guards to exist early.
```
guard let deviceSlot = _activeTouchIds.firstIndex(of: -1) else { continue } 
```

I won't point them our separately, but this would apply to the added if statements below as well.

Reply:
Good catch! Fixed 👍🏻